cumulusci/utils/yaml/cumulusci_yml.py

```diff
@@ -1,4 +1,276 @@
+from cumulusci.utils.yaml.safer_loader import load_yaml_data
+
+default_logger = getLogger(__name__)
+
+
+NBSP = "\u00A0"
+
+pattern = re.compile(r"^\s*[\u00A0]+\s*", re.MULTILINE)
+
+
+#  type aliases
+PythonClassPath = str
+URL = str
+
+
+class FlowReference(CCIDictModel):
+    flow: str = None
+    options: Dict[str, Any] = {}
+    ignore_failure: bool = None  # is this allowed?
+    when: str = None  # is this allowed?
+    ui_options: Dict[str, Any] = {}
+
+
+class TaskReference(CCIDictModel):
+    task: str
+    options: Dict[str, Any] = {}
+    ui_options: Dict[str, Any] = {}
+    ignore_failure: bool = None
+    when: str = None  # is this a documented feature?
+    # classpath, description allowed?
+
+
+class Task(CCIDictModel):
+    name: str = None  # is this real or a typo
+    class_path: str = None
+    description: str = None
+    options: Dict[str, Any] = None
+    group: str = None
+    ui_options: Dict[str, Any] = None
+
+
+class Flow(CCIDictModel):
+    description: str = None
+    steps: Dict[str, Union[FlowReference, TaskReference]]
+    group: str = None
+
+
+class Package(CCIDictModel):
+    name: Optional[str] = None
+    name_managed: Optional[str] = None
+    namespace: Optional[str] = None
+    install_class: str = None
+    uninstall_class: str = None
+    api_version: str = None
+
+
+class Test(CCIDictModel):
+    name_match: str
+
+
+class Parser(CCIDictModel):
+    class_path: PythonClassPath
+    title: str
+
+
+class ReleaseNotes(CCIDictModel):
+    parsers: Dict[int, Parser]  # should probably be a list
+
+
+class Git(CCIDictModel):
+    repo_url: str = None
+    default_branch: str = None
+    prefix_feature: str = None
+    prefix_beta: str = None
+    prefix_release: str = None
+    push_prefix_sandbox: str = None
+    push_prefix_production: str = None
+    release_notes: ReleaseNotes = None
+    two_gp_context: str = Field(None, alias="2gp_context")
+
+
+class ApexDoc(CCIDictModel):
+    homepage: str = None  # can't find these in the CCI Python code
+    banner: str = None
+    branch: str = None
+    repo_dir: Path = None
+
+
+class Check(CCIDictModel):
+    when: str = None
+    action: str = None
+    message: str = None
+
+
+class Plan(CCIDictModel):  # MetaDeploy plans
+    title: str
+    description: str = None
+    tier: str = None
+    slug: str = None
+    is_listed: bool = None
+    steps: Dict[str, Union[FlowReference, TaskReference]]
+    checks: List[Check] = None
+    group: str = None
+    error_message: str = None
+    error_message: str = None
+    post_install_message: str = None
+    preflight_message: str = None
+
+
+class Project(CCIDictModel):
+    name: Optional[str]
+    package: Package = None
+    test: Test = None
+    git: Git = None
+    dependencies: List[Dict[str, URL]] = None  # TODO
+    apexdoc: ApexDoc = None
+    source_format: Literal["sfdx", "mdapi"] = "mdapi"
+
+
+class ScratchOrg(CCIDictModel):
+    config_file: Path
+    days: int = None
+    namespaced: str = None
```

Comment:
Add setup_flow: str = None

(It's not used by CumulusCI yet but it can be used by Metecho)